src/xdp-session.c

```diff
@@ -311,6 +311,20 @@ xdp_session_authorize_callback (GDBusInterfaceSkeleton *interface,
   return TRUE;
 }
 
+static gboolean
+is_valid_token(const char *token)
+{
+  int i;
+
+  for (i = 0; token[i]; i++)
+    {
+      if (!g_ascii_isalnum(token[i]) && token[i] != '_')
```

Comment:
Is `_` a valid token? Doesn't gdbus already provide something like this?

Reply:
If this is being used in object paths, https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names is the canonical specification.

gdbus probably doesn't have a function for "is this a valid object-path *component*?", though, just "is this a valid complete object path?" (which is `g_variant_is_object_path()`).

Reply:
> Is `_` a valid token?

If our only functional requirement is "can we append it to `/blah/blah/` and get a valid object path?" then yes it is.